src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -260,9 +252,11 @@ delete_subject_version(server::request_t rq, server::reply_t rp) {
     rq.req.reset();
 
     auto version = invalid_schema_version;
+    auto versions = co_await rq.service().schema_store().get_versions(
+      sub, include_deleted::yes);
+    auto last_version = versions.size() == 1;
```

Comment:
Does this have the correct behaviour for multiple deletes of "latest"? Should it delete, say 5 and then 4?

Reply:
Hmm, confluent's API docs say that latest is only valid for soft deletes, and I think in that case we're meant to soft-delete the most recent one that is not already marked soft deleted (which wasn't exactly what we were doing).  I've updated the logic to match that.